### ISSUE TITLE
Please Don't Rely on Colors in the Interface

### ISSUE THREAD

**Issue by archolewa** [sil-quirk/sil-q#7]
So I've been messing around with both Sil and Sil-Q a bit over the past few days. They're *very* cool games. In particular I adore the song-based, buff/debug only magic. One thing that really bugs me though is that they rely solely on colors on some interfaces. In particular, the game changes the color of the statistic/skill the player currently has selected when they're increasing skills and applying experience points.

I'm not a big fan of color. At best I find it distracting, and often it actively inhibits my ability to play a game or read text (i.e. if the text is dark blue on a black terminal), so I have my $TERM set to xterm-mono. Unfortunately, in Sil-q this means that I don't know which skill (or statistic) is currently highlighted when I'm improving my character. 

It would be wonderful if the interface was enhanced to make colors optional. Maybe a cursor pointing at the current skill, or even just bolding the number. Or providing additional keys to modify statistics like there is on the color modification menu. For example s/S could perhaps be used to decrease/increase strength. Then players don't have to care which statistic is currently highlighted.

**Comment by sil-quirk** [comment]
Hi archolewa. Thanks for the input. I understand where you're coming from, though so far this is the first time I've heard this request.

My concern is that there are quite a few menus that use colour to some degree (new game, stats, skills, abilities, smithing, options, etc) and certainly for distinguishing usable from greyed out I suspect adding optional markers won't make the interface any prettier. Could we consider some sort of compromise e.g. a compilation flag that enabled a marker for the most annoying interfaces (stats, skills)? Bolding the number / ability would probably be my preference, if that turns out to be reasonably straightforward, but something like a >  < indicator might work also.

I'd rather not go down the additional keys route - learning the keys in a roguelike is a painful enough endeavour as is, and using s/S to increase/decrease Strength takes up the obvious key for Stealth, Smithing and Song (which you will likely modify many more times over the course of the game).

**Comment by archolewa** [comment]
I think a flag for the most annoying interfaces is totally fine. The only interface I've found so far that absolutely *needs* color is the stat/skill-increasing interface, and that's just to know which stat/skill is currently highlighted. Everything else I've interacted with lets me make a choice by pressing a key (i.e. `a` to smith a sword, `p` to learn a song) rather than scrolling up and down. 

I don't even need things to be greyed out. I'll know I can't do something (select a skill, smith an item) if I try to select it and I get a beep, or nothing happens. Plus, there's usually other feedback that tells me if I can do something or not. For example, skills list their requirements, so if I don't meet the requirements, I can't learn that skill.

Bolding the currently selected skill is totally fine. I'm not married to any particular method. I just want to know which skill I'm trying to improve without having to (*gasp* *horror*) turn on colors.

I *may* be a little...peculiar.
